src/main/java/org/mockito/internal/util/reflection/FieldSetter.java

```diff
@@ -6,21 +6,56 @@
 
 import java.lang.reflect.Field;
 
-public class FieldSetter {
-
-    private FieldSetter(){}
-
-    public static void setField(Object target, Field field,Object value) {
-        AccessibilityChanger changer = new AccessibilityChanger();
-        changer.enableAccess(field);
-        try {
-            field.set(target, value);
-        } catch (IllegalAccessException e) {
-            throw new RuntimeException("Access not authorized on field '" + field + "' of object '" + target + "' with value: '" + value + "'", e);
-        } catch (IllegalArgumentException e) {
-            throw new RuntimeException("Wrong argument on field '" + field + "' of object '" + target + "' with value: '" + value + "', \n" +
+public final class FieldSetter {
+
+    private FieldSetter() {}
+
+    /**
+     * Set value of fields, except static final (use {@link #setAnyField(Object, Field, Object)} for that).
+     */
+    public static void setField(Object target, Field field, Object value) {
+        // keep synchronization in sync with setAnyField method!
+        synchronized (field) {
```

Comment:
I'm not sure this synchronization needed since this call is wrapped by `setAnyField` which is supposed to handle static fields as well.

Reply:
TIL: Foo.class.getDeclaredField("bar") returns a new Field instance for every call
=> synchronizing is probably not worth it.
I'll remove this